js/pivx_shield.ts

```diff
@@ -273,6 +282,9 @@ export class PIVXShield {
       shieldData.lastProcessedBlock,
       shieldData.commitmentTree,
     );
+    pivxShield.mapNullifierNote = new Map(
+      Object.entries(shieldData.mapNullifierNote),
```

Comment:
This needs to handle being undefined
```suggestion
      Object.entries(shieldData.mapNullifierNote ?? {}),
```